packages/web/src/pages/chat-page/components/ChatMessageListItem.tsx

```diff
@@ -62,6 +66,19 @@ export const ChatMessageListItem = (props: ChatMessageListItemProps) => {
   const senderUserId = decodeHashId(message.sender_user_id)
   const isAuthor = userId === senderUserId
   const links = find(message.message)
+  const {
+    kind,
+    track,
+    trackStatus,
+    trackError,
+    playlist,
+    playlistStatus,
+    playlistError
+  } = useTrackOrPlaylist(message)
```

Comment:
We should check with product, but I _think_ this should have the same behavior as other link previews. I don't think it only shows the track tile if the whole message is a track link, but rather it shows the track tile if there exists a track link in the message.

If that's what product says, I think we should do a switch/if-else on the _links_ and do the same sort of thing:

```
if link is track link
  -> render chat track tile
if link is playlist link
  -> render chat playlist tile
else
  -> render link preview
```

Reply:
It also seems like we can get rid of `useTrackOrPlaylist`? Can `ChatMessageTrack` take in the permalink and call `useGetTrackByPermalink` and `ChatMessagePlaylist` do the same with `useGetPlaylistByPermalink`?

Reply:
agreed. wanted to build it like this first and tweak logic later (which is now lol), but great callout and thx for confirming.  going to likely do this in another PR, which is going to include some of my existing changes, to make this PR smaller and make it easier for you to re-review